types/src/system/auction/mod.rs

```diff
@@ -548,35 +548,24 @@ pub trait Auction:
                     .ok_or(Error::ArithmeticOverflow)?
             };
 
-            let delegator_total_stake: U512 = recipient
-                .delegator_total_stake()
-                .ok_or(Error::ArithmeticOverflow)?;
-
-            let delegators_part: Ratio<U512> = {
-                let commission_rate = Ratio::new(
-                    U512::from(*recipient.delegation_rate()),
-                    U512::from(DELEGATION_RATE_DENOMINATOR),
-                );
-                let reward_multiplier: Ratio<U512> = Ratio::new(delegator_total_stake, total_stake);
-                let delegator_reward: Ratio<U512> = total_reward
-                    .checked_mul(&reward_multiplier)
-                    .ok_or(Error::ArithmeticOverflow)?;
-                let commission: Ratio<U512> = delegator_reward
-                    .checked_mul(&commission_rate)
-                    .ok_or(Error::ArithmeticOverflow)?;
-                delegator_reward
-                    .checked_sub(&commission)
-                    .ok_or(Error::ArithmeticOverflow)?
-            };
+            let reward_per_stake: Ratio<U512> = Ratio::<U512>::new(
+                (DELEGATION_RATE_DENOMINATOR - recipient.delegation_rate()).into(),
```

Comment:
This looks correct, but it might be a tad convoluted, since it does (d-r)/(d*S) * R, which is just R/S * (1 - r/d), is there any advantage to this?

Reply:
The intermediate values are smaller and so it reduces the risk of integer overflow.